tftpy/TftpStates.py

```diff
@@ -182,10 +182,14 @@ def handleDat(self, pkt):
                 log.warn("There is no block zero!")
                 self.sendError(TftpErrors.IllegalTftpOp)
                 raise TftpException, "There is no block zero!"
-            log.warn("Dropping duplicate block %d" % pkt.blocknumber)
-            self.context.metrics.add_dup(pkt)
-            log.debug("ACKing block %d again, just in case", pkt.blocknumber)
-            self.sendACK(pkt.blocknumber)
+            if pkt.blocknumber == self.context.next_block - 1:
+                log.warn("Received duplicate DAT for "
+                         "previous block %d. Resending last packet."
+                         % pkt.blocknumber)
```

Comment:
Super minor.  For log statements, you can have a format string an instead of using `%` you can pass the format arguments as arguments to the logging function.  This defers the format operation until the time at which the log statement is emitted (if it is emitted).  Just reduces work if the statement is not going to be logged.

Here's what this would look like using that methodology:

```
            if pkt.blocknumber == self.context.next_block - 1:
                log.warn("Received duplicate DAT for "
                         "previous block %d. Resending last packet.",
                         pkt.blocknumber)
```